File: project-03-realtime-dashboard/code-java/src/main/java/com/example/demo/dto/ProductSalesRankDto.java
package com.example.demo.dto;

import java.math.BigDecimal;
import java.time.LocalDate;

public record ProductSalesRankDto(
    LocalDate saleDate,
    Long productId,
    String category,
    BigDecimal revenue,
    BigDecimal previousRevenue,
    Integer rank
) {}

File: project-03-realtime-dashboard/code-kotlin/src/main/kotlin/com/example/demo/dto/ProductSalesRankDto.kt
package com.example.demo.dto

import java.math.BigDecimal
import java.time.LocalDate

data class ProductSalesRankDto(
    val saleDate: LocalDate,
    val productId: Long,
    val category: String,
    val revenue: BigDecimal,
    val previousRevenue: BigDecimal?,
    val rank: Int
)
